src/controllers/species.controller.ts

```diff
@@ -61,23 +66,40 @@ export class SpeciesController {
     filter?: Filter<Species> & { fields?: { captureCount?: Boolean } },
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
   ): Promise<any[]> {
-    console.log('get /species filter --> ', filter ? filter.where : null);
     // Only include active species
     filter = { ...filter, where: { ...filter?.where, active: true } };
+    const speciesList = await this.speciesRepository.find(filter);
 
+    if (this.requestHasUser(this.request)) {
+      //CheckUserAccess
+      const userRoles: string[] = this.request.user.roleNames;
```

Comment:
We should be checking user _policies_, not _roles_ (`this.request.user.policy.policies`).
See `src/js/auth.js` for the supported policies and how these checks are implemented (`isAuth()`).

Reply:
There were no species manager policy in Auth.js.
below are list of policies. which should i allow?

` SUPER_PERMISSION: 'super_permission',
  LIST_USER: 'list_user',
  MANAGER_USER: 'manager_user',
  LIST_TREE: 'list_tree',
  APPROVE_TREE: 'approve_tree',
  LIST_PLANTER: 'list_planter',
  MANAGE_PLANTER: 'manage_planter',`

Reply:
Just `SUPER_PERMISSION` is fine. We're not going to add the `'species_manager'` policy to the legacy API.

Reply:
Done.